core/app/models/spree/product/slugs.rb

```diff
@@ -0,0 +1,101 @@
+module Spree
+  class Product < Spree.base_class
+    module Slugs
+      extend ActiveSupport::Concern
+
+      included do
+        extend FriendlyId
+        include Spree::TranslatableResourceSlug
+
+        translates :slug
+        friendly_id :slug_candidates, use: [:history, :slugged, :scoped, :mobility], scope: spree_base_uniqueness_scope, slug_limit: 255
+        acts_as_paranoid
```

Comment:
@resool let's not move this outside of the main model, this is not slug specific logic

`acts_as_paranoid` should stay in product model main file